src/preprocessing/particle_packing/nhs_faces.jl

```diff
@@ -0,0 +1,263 @@
+        for cell in cell_grid(face, geometry, neighborhood_search)
+            if cell_intersection(face, geometry, cell, neighborhood_search)
+                if haskey(cell_list, cell) && !(face in cell_list[cell])
+                    # Add face to corresponding cell
+                    append!(cell_list[cell], face)
+                else
+                    # Create cell
+                    cell_list[cell] = [face]
+                end
+            end
+        end
+    end
+
+    empty!(neighbor_iterator)
+
+    min_cell = PointNeighbors.cell_coords(geometry.min_corner, neighborhood_search) .- pad
+    max_cell = PointNeighbors.cell_coords(geometry.max_corner, neighborhood_search) .+ pad
+
+    # Merge all lists of faces in the neighboring cells into one iterator
+    face_ids = Int[]
+    for cell_runner in meshgrid(min_cell, max_cell)
+        resize!(face_ids, 0)
+        for neighbor in PointNeighbors.neighboring_cells(cell_runner, neighborhood_search,
+                                                         search_radius)
+            append!(face_ids, faces_in_cell(Tuple(neighbor), neighborhood_search))
+        end
+
+        unique!(face_ids)
+
+        if isempty(face_ids)
+            continue
+        end
+
+        neighbor_iterator[cell_runner] = copy(face_ids)
+    end
+
+    return neighborhood_search
+end
+
+function cell_intersection(edge, geometry, cell,
+                           neighborhood_search::FaceNeighborhoodSearch{2})
+    (; cell_size) = neighborhood_search
+
+    v1, v2 = face_vertices(edge, geometry)
+
+    # Check if one of the vertices is inside cell
+    cell == PointNeighbors.cell_coords(v1, neighborhood_search) && return true
+    cell == PointNeighbors.cell_coords(v2, neighborhood_search) && return true
+
+    # Check if line segment intersects cell
+    min_corner = SVector(cell .* cell_size...)
+    max_corner = min_corner + SVector(cell_size...)
+
+    ray_direction = v2 - v1
+
+    return ray_intersection(min_corner, max_corner, v1, ray_direction)
+end
+
+function cell_intersection(triangle, geometry, cell,
+                           neighborhood_search::FaceNeighborhoodSearch{3})
+    (; cell_size) = neighborhood_search
+
+    v1, v2, v3 = face_vertices(triangle, geometry)
+
+    # Check if one of the vertices is inside cell
+    cell == PointNeighbors.cell_coords(v1, neighborhood_search) && return true
+    cell == PointNeighbors.cell_coords(v2, neighborhood_search) && return true
+    cell == PointNeighbors.cell_coords(v3, neighborhood_search) && return true
+
+    # Check if line segment intersects cell
+    min_corner = SVector(cell .* cell_size...)
+    max_corner = min_corner + SVector(cell_size...)
+
+    ray_direction1 = v2 - v1
+    ray_intersection(min_corner, max_corner, v1, ray_direction1) && return true
+
+    ray_direction2 = v2 - v3
+    ray_intersection(min_corner, max_corner, v1, ray_direction2) && return true
```

Comment:
Also shouldn't this be v3?
ray_intersection(min_corner, max_corner, v3, ray_direction2)